### ISSUE TITLE
umi correction issues

### ISSUE THREAD

**Issue by cusanovich** [Hoohm/CITE-seq-Count#45]
I'm not sure if this is an issue particular to my system, but I'm getting errors in the `correct_cells`  and `correct_umis` functions when running the program on some hash data.

For cell barcodes, the program reports: "Could not find a good local minima for correction." When I dig into it a little further, it appears that the `correct_cells` function uses `umi_methods.getCellWhitelist()`, which my system doesn't recognize. Instead, umi_tools seems to have a `whitelist_methods.getCellWhitelist()` function. Is this an incompatibility between umi_tools version (I'm using 1.0.0) and CITE-seq-Count version (I'm using 1.4.1)?

For umis, I  get the error: "TypeError: __call__() takes 3 positional arguments but 4 were given". When I dig into this a little more, it seems that `network.UMIClusterer()` (from umi_tools again) is the culprit. As far as I can tell, you are feeding it the umis as keys, a dictionary of umis with counts and a threshold. However, the function seems to expect only the dictionary and the threshold on my machine.

If I fix both issues in the source code, the program seems to run fine. I assume that this is a versioning issue on my machine since I don't see anyone else noting it, but thought I'd share anyway.

**Comment by zakiF** [comment]
Hi @cusanovich ,

Just to mention I am getting the same issue as well. Part of the error message is as follows ;

```
  File "/Users/CITE-seq-Count/cite_seq_count/processing.py", line 190, in correct_umis
    collapsing_threshold)
TypeError: __call__() takes 3 positional arguments but 4 were given
```

I am using CITE-seq-Count v1.4.2 (from devel) and UMI-tools version: 1.0.0


A quick fix was as per your suggestion, commenting line 188 in the processing.py did the job and allowed the program to run.

```
UMIclusters = umi_clusters(
                    #final_results[cell_barcode][TAG].keys(),  # Commented this line
                    final_results[cell_barcode][TAG],
                    collapsing_threshold)
```

Thanks for the suggestion.

**Comment by Hoohm** [comment]
Hello,
the issue comes from the updated umi_tools and changes they made.
This is now fixed in 1.4.2.

**Comment by Hoohm** [comment]
Fixed in 1.4.2, closing this
